lib/proxy_api/resource.rb

```diff
@@ -63,15 +63,22 @@ def parse(response)
       false
     end
 
+    def escape_path(path)
+      parsed_path = URI.parse(path)
+      query = parsed_path.query
+      path = "#{parsed_path.path}?#{URI.encode_www_form(CGI.parse(query))}" unless query.nil?
+      path
+    end
```

Comment:
This doesn't look correct to me. While I don't expect a problem with correct input, it doesn't help anything. With incorrect input you actually get a very different output.

For example:
```
[1] pry(main)> path = URI.parse('http://example.com/?parameter1=a text with & in it')
=> #<URI::HTTP http://example.com/?parameter1=a%20text%20with%20&%20in%20it>
[2] pry(main)> path.query
=> "parameter1=a%20text%20with%20&%20in%20it"
[3] pry(main)> require 'cgi'
=> true
[4] pry(main)> CGI.parse(path.query)
=> {"parameter1"=>["a text with "], " in it"=>[]}
[5] pry(main)> URI.encode_www_form(CGI.parse(query))
NameError: undefined local variable or method `query' for main:Object
from (pry):5:in `__pry__'
[6] pry(main)> URI.encode_www_form(CGI.parse(path.query))
=> "parameter1=a+text+with+&"
```

Now this is garbage input, but that makes me think we should stop escaping path and requiring the caller to properly escape. That's the only place where it can really be done reliably.